src/ccutil/unicharmap.cpp

```diff
@@ -39,7 +39,7 @@ UNICHAR_ID UNICHARMAP::unichar_to_id(const char* const unichar_repr,
   assert(length > 0 && length <= UNICHAR_LEN);
 
   int index = 0;
-  if (index >= length || unichar_repr[index] == '\0') return INVALID_UNICHAR_ID;
+  if (unichar_repr[index] == '\0') return INVALID_UNICHAR_ID;
```

Comment:
That won't handle the case `length <= 0` in release code (which removes the `assert`).

Reply:
It should probably be changed to `length <= 0 || ...` since that's the real bug and not a bug caused by the `index` variable
@stweil Thoughts?